apps/worker/src/request_processor/request.processor.service.ts

```diff
@@ -37,6 +39,15 @@ export class RequestProcessorService extends BaseConsumer {
   async process(job: Job<ProviderGraphUpdateJob, any, string>): Promise<void> {
     this.logger.log(`Processing job ${job.id} of type ${job.name}`);
     try {
+      const lastProcessedDsnpId = await this.cacheManager.get(QueueConstants.LAST_PROCESSED_DSNP_ID_KEY);
+      if (lastProcessedDsnpId && lastProcessedDsnpId === job.data.dsnpId) {
+        const blockDelay = SECONDS_PER_BLOCK * MILLISECONDS_PER_SECOND;
+        this.logger.debug(`Delaying processing of job ${job.id} for ${blockDelay}ms`);
+        // eslint-disable-next-line no-await-in-loop
+        await new Promise((r) => {
+          setTimeout(r, blockDelay);
+        });
```

Comment:
In cases where requests are sent back to back and first request might not have yet went through the tx on chain, we would wait a block to ensure the next fetch of same user has updated hash,

this would prevent the post tx failures and save some capacity for provider